src/gatsby/gatsby-node/createResolvers.ts

```diff
@@ -0,0 +1,68 @@
+/* eslint-disable @typescript-eslint/no-var-requires */
+import {
+  CreateResolversArgs,
+  GatsbyNode,
+} from 'gatsby';
+
+const genMDX = require('gatsby-plugin-mdx/utils/gen-mdx');
+
+import getTocItems from '../../utils/get-toc-items';
+
+export const createResolvers: GatsbyNode['createResolvers'] = async(
+  args: CreateResolversArgs,
+): Promise<any> => {
+  const { createResolvers } = args;
+  const {
+    store,
+    pathPrefix,
+    getNode,
+    getNodes,
+    cache,
+    reporter,
+    actions,
+    schema,
+    ...helpers
+  } = args;
+
+  const plugins = args.store.getState().flattenedPlugins;
+
+  const mdxPlugin = plugins.find(
+    (plugin: any) => plugin.name === 'gatsby-plugin-mdx'
+  );
+
+  const processMDX = ({ node }: any) =>
+    genMDX({
+      actions,
```

Comment:
If the above cleanup is implemented, you could reduce this to `{node, options: mdxPlugin.pluginOptions, store, ...helpers}`